doc/python/categorical-colors.md

```diff
@@ -0,0 +1,231 @@
+             color_discrete_sequence=["red", "green", "blue", "goldenrod", "magenta"],
+             title="Explicit color sequence"
+            )
+
+fig.show()
+```
+
+**_Warning_**: If your color sequence is has fewer colors than the number of unique values in the column you are mapping to `color`, the colors will cycle through and repeat, possibly leading to ambiguity:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+fig = px.scatter(df, x="total_bill", y="tip", color="day",
+             color_discrete_sequence=["red", "blue"],
+             title="<b>Ambiguous!</b> Explicit color sequence cycling because it is too short"
+            )
+
+fig.show()
+```
+
+### Directly Mapping Colors to Data Values
+
+The example above assigned colors to data values on a first-come-first-served basis, but you can directly map colors to data values if this is important to your application with `color_discrete_map`. Note that this does not change the order in which values appear in the figure or legend, as can be controlled below:
+
+```python
+import plotly.express as px
+df = px.data.gapminder().query("year == 2007")
+fig = px.bar(df, y="continent", x="pop", color="continent", orientation="h", hover_name="country",
+             color_discrete_map={
+                "Europe": "red",
+                "Asia": "green",
+                "Americas": "blue",
+                "Oceania": "goldenrod",
+                "Africa": "magenta"},
+             title="Explicit color mapping")
+
+fig.show()
+```
+
+### Controlling Categorical Order
+
+Plotly Express lets you specify an ordering over categorical variables with `category_orders`, which will apply to colors and legends as well as symbols, [axes](/python/axes/) and [facets](/python/facet-plots/). This can be used with either `color_discrete_sequence` or `color_discrete_map`.
+
+```python
+import plotly.express as px
+df = px.data.gapminder().query("year == 2007")
+fig = px.bar(df, y="continent", x="pop", color="continent", orientation="h", hover_name="country",
+             color_discrete_map={
+                "Europe": "red",
+                "Asia": "green",
+                "Americas": "blue",
+                "Oceania": "goldenrod",
+                "Africa": "magenta"},
+             category_orders={"continent": ["Oceania", "Europe", "Asia", "Africa", "Americas"]},
+             title="Explicit color sequence with explicit ordering"
+            )
+
+fig.show()
+```
+
+```python
+import plotly.express as px
+df = px.data.gapminder().query("year == 2007")
+fig = px.bar(df, y="continent", x="pop", color="continent", orientation="h", hover_name="country",
+             color_discrete_sequence=["red", "green", "blue", "goldenrod", "magenta"],
+             category_orders={"continent": ["Oceania", "Europe", "Asia", "Africa", "Americas"]},
+             title="Explicit color mapping with explicit ordering"
```

Comment:
I'm a bit confused about the titles of these two examples. Shouldn't the previous one be "Explicit color mapping with explicit ordering" and this one "Explicit color sequence with explicit ordering"?